### ISSUE TITLE
Table merges end up shrinking table columns

### ISSUE THREAD

**Issue by dkrahn** [ckeditor/ckeditor5#1468]
## Is this a bug report or feature request? (choose one)

🐞 Bug report

## 💻 Version of CKEditor

11.2.0 (Tested on https://ckeditor5.github.io/)

## 📋 Steps to reproduce

1. Try to create a table with the same layout as the TableWalker documentation(showed below)
```
+----+----+----+----+----+----+
| 00      | 02 | 03      | 05 |
|         +----+----+----+----+
|         | 12      | 14 | 15 |
|         +----+----+----+----+
|         | 22                |
|----+----+                   +
| 31 | 32 |                   |
+----+----+----+----+----+----+
```

## ✅ Expected result
Have the desired table layout
```
+----+----+----+----+----+----+
| 00      | 02 | 03      | 05 |
|         +----+----+----+----+
|         | 12      | 14 | 15 |
|         +----+----+----+----+
|         | 22                |
|----+----+                   +
| 31 | 32 |                   |
+----+----+----+----+----+----+
```

## ❎ Actual result
```
+----+----+----+----+----+
| 00      | 02 | 03 | 05 |
|         +----+----+----+
|         | 12 | 14 | 15 |
|----+----+----+----+----+
| 31 | 32 | 22           |
+----+----+----+----+----+
```

## 📃 Other details that might be useful
Did not find a way to accomplish it, any order of merges give the same result.

**Comment by jodator** [comment]
Well that's unfortunate but it seems that The example might be not valid (I'd have to confirm that with specs).

This is how this happen:
![peek 2019-01-17 09-29](https://user-images.githubusercontent.com/247363/51305320-cfe7a180-1a3a-11e9-8da4-ba75bccfddb5.gif)

This is how the DOM is rendered in FF:
![peek 2019-01-17 09-30](https://user-images.githubusercontent.com/247363/51305339-db3acd00-1a3a-11e9-8a29-4d2677cad22d.gif)

And this is the markup (without additional attributes and td contents):
```html
<table>
	<tbody>
		<tr>
			<td rowspan="3" colspan="2"></td>
			<td></td>
			<td colspan="2"></td>
			<td></td>
		</tr>
		<tr>
			<td colspan="2"></td>
			<td></td>
			<td></td>
		</tr>
		<tr>
			<td colspan="4" rowspan="2"></td>
		</tr>
		<tr>
			<td></td>
			<td></td>
		</tr>
	</tbody>
</table>
```

Below is a table rendered in Chrome with only user-agent styles (+ borders for td). You can see that problematic cells (`12` and `22`) are render slightly off:
![selection_159](https://user-images.githubusercontent.com/247363/51305879-5486ef80-1a3c-11e9-9386-9245b1fdeb19.png)

And similar thing with CKEditor 4:
![selection_161](https://user-images.githubusercontent.com/247363/51305984-92841380-1a3c-11e9-947d-4834d968d486.png)

So without much digging into I think that at least we should update the example in the docs. Fortunately the markup is correct but the problem lies in how browsers renders such markup.

**Comment by oleq** [comment]
Does that happen in all browsers?

**Comment by jodator** [comment]
@oleq Firefox & Chrome - I didn't test Edge though.

**Comment by oleq** [comment]
OK, so maybe let's not use a confusing content HTML in the docs.

**Comment by Reinmar** [comment]
I'm lost – what needs to be done to resolve this ticket?

**Comment by jodator** [comment]
@Reinmar The sample needs to be updated so it will provide "visually correct" table. As the sample in the docs produces "visually incorrect" table as browsers collapse cells/rows and thus it looks incorrect. But this is only visual - the HTML is valid.

tl;dr: provide other sample in that docs that will not create confusing effect in the browser

**Comment by dkrahn** [comment]
The example could be just add a new line and column with no row/col span.
```
+----+----+----+----+----+----+----+
| 00      | 02 | 03      | 05 | 06 |
|         +----+----+----+----+----+
|         | 12      | 14 | 15 | 16 |
|         +----+----+----+----+----+
|         | 22                | 26 |
|----+----+                   +----+
| 31 | 32 |                   | 36 |
+----+----+----+----+----+----+----+
| 40 | 41 | 42 | 43 | 44 | 45 | 46 |
+----+----+----+----+----+----+----+
```
This will make the browser not collapse the table.

**Comment by jodator** [comment]
@dkrahn thanks! This will land in the docs for upcoming release.

**Comment by jodator** [comment]
After some fiddling I went with:

```
+----+----+----+----+----+----+
| 00      | 02 | 03 | 04 | 05 |
|         +----+----+----+----+
|         | 12      | 14 | 15 |
|         +----+----+----+    +
|         | 22           |    |
|----+----+----+----+----+    +
| 30 | 31 | 32 | 33 | 34 |    |
+----+----+----+----+----+----+
```
As this table is the same size as previous and illustrates how walker skips rows/columns. Also is nice in the rendered table:
![selection_195](https://user-images.githubusercontent.com/247363/52860500-e9f6cb80-312f-11e9-8240-dfe422c21270.png)

**Comment by jodator** [comment]
Closed by: https://github.com/ckeditor/ckeditor5-table/commit/b97bc1200cbe9b9be0726dc2536ba662b946c2b7.
